src/renderers/webgl/WebGLPrograms.js

```diff
@@ -134,10 +135,14 @@ function WebGLPrograms( renderer, extensions, capabilities ) {
 
 		var parameters = {
 
+			isWebGL2: capabilities.isWebGL2,
+
 			shaderID: shaderID,
 
 			precision: precision,
-			isWebGL2: capabilities.isWebGL2,
+
+			instancing: object.isInstancedMesh === true,
```

Comment:
Are there other cases where reusing a single material on multiple objects would create multiple shader programs? Wondering if this should require a `material.instancing` flag or something, similar to `.skinning`.

Reply:
Oh, I have not tested this.

Reply:
I'll leave that for the next cycle.

Reply:
I've pointed out the similar stuff [here](https://github.com/mrdoob/three.js/pull/17505#issuecomment-532012299), probably material can't be shared between regular mesh and instanced mesh.

And wondering if material can have multiple shader programs depending on some conditions, like `SkinnedMesh`, `InstancedMesh`, or so on. We can remove `.skinning` or such other properties from material and material can be shared between them.